sled-agent/src/rack_setup/service.rs

```diff
@@ -1363,14 +1370,34 @@ impl ServiceInner {
 
         // Wait until time is synchronized on all sleds before proceeding.
         rss_step.update(RssStep::WaitForTimeSync);
-        let sled_addresses: Vec<_> = sled_plan
-            .sleds
-            .values()
-            .map(|initialization_request| {
-                get_sled_address(initialization_request.body.subnet)
+        let ntp_addresses: Vec<_> = service_plan
+            .services
+            .iter()
+            .flat_map(|(_, sled_config)| {
+                sled_config.zones.iter().filter_map(|zone_config| {
+                    match &zone_config.zone_type {
+                        BlueprintZoneType::BoundaryNtp(
+                            blueprint_zone_type::BoundaryNtp {
+                                address, ..
+                            },
+                        ) => {
+                            let mut ntp_admin_addr = *address;
+                            ntp_admin_addr.set_port(NTP_ADMIN_PORT);
+                            Some(ntp_admin_addr)
+                        }
+                        BlueprintZoneType::InternalNtp(
+                            blueprint_zone_type::InternalNtp { address },
+                        ) => {
+                            let mut ntp_admin_addr = *address;
+                            ntp_admin_addr.set_port(NTP_ADMIN_PORT);
+                            Some(ntp_admin_addr)
+                        }
```

Comment:
Since both of these arms bind a field named `address` with the same type, I think you can combine them?

```rust
                        BlueprintZoneType::BoundaryNtp(
                            blueprint_zone_type::BoundaryNtp {
                                address, ..
                            },
                        ) 
                        | BlueprintZoneType::InternalNtp(
                            blueprint_zone_type::InternalNtp { address },
                        ) => {
                            let mut ntp_admin_addr = *address;
                            ntp_admin_addr.set_port(NTP_ADMIN_PORT);
                            Some(ntp_admin_addr)
                        }
```

Reply:
Good catch, updated